.github/workflows/build-in-devcontainer.yaml

```diff
@@ -119,36 +120,67 @@ jobs:
 
       - name: Check if repo has devcontainer
         env:
+          ARCH: ${{ matrix.ARCH }}
           CUDA_VER: ${{ matrix.CUDA_VER }}
           PACKAGER: ${{ matrix.PACKAGER }}
+          RUN_ID: ${{ github.run_id }}
+          RUN_ATTEMPT: ${{ github.run_attempt }}
         run: |
-          echo "REPOSITORY=$(basename "$(pwd)")" | tee -a "${GITHUB_ENV}"
-          if test -f ".devcontainer/cuda${CUDA_VER}-${PACKAGER}/devcontainer.json"; then
-            echo "HAS_DEVCONTAINER=true" >> "${GITHUB_ENV}"
-          else
-            echo "HAS_DEVCONTAINER=false" >> "${GITHUB_ENV}"
+          HAS_DEVCONTAINER=false
+          if test -f "repo/.devcontainer/cuda${CUDA_VER}-${PACKAGER}/devcontainer.json"; then
+            HAS_DEVCONTAINER=true
```

Comment:
If this is false because there’s no devcontainer, what does the “build devcontainer” workflow do? How does the conditional help us?

Reply:
> _If there's no devcontainer, we want to do nothing without failing the build._

Why would we not want to fail the build in that case?

I don't think we should do this... it could silently cover up actual problems. For example, if any repo deviates from the standard naming convention we have (`cuda${CUDA_VER}-${PACKAGER}/devcontainer.json`), today you get a big loud CI failure telling you about that.

With this check, that'd turn into this job just silently being a no-op, which might be missed.

Repos without devcontainers should just not use this workflow, and this workflow should fail outright if what it needs isn't found.

Reply:
We talked offline about this... this is not new behavior, it's only showing up here because of the change to git cloning that now writes the repo to `repo/` (changing the file path passed to `test -f`).

Approving this, but I do think we should remove this `HAS_DEVCONTAINER=false` behavior in a follow-up.

Reply:
Great. I agree we should fail loudly here, and remove this in a follow-up.